In [9]:
import pandas as pd
def read_lines(path):
    return [line.rstrip('\n') for line in open(path,encoding='utf-8')]

train= pd.read_csv('korean-hate-speech/labeled/train.tsv' ,sep='\t')
dev= pd.read_csv('korean-hate-speech/labeled/dev.tsv' ,sep='\t')

train_news_title = read_lines('korean-hate-speech/news_title/train.news_title.txt')
dev_news_title = read_lines('korean-hate-speech/news_title/dev.news_title.txt')

In [13]:
train

comments  contain_gender_bias  \
0     (현재 호텔주인 심정) 아18 난 마른하늘에 날벼락맞고 호텔망하게생겼는데 누군 계속...                False   
1     ....한국적인 미인의 대표적인 분...너무나 곱고아름다운모습...그모습뒤의 슬픔을...                False   
2     ...못된 넘들...남의 고통을 즐겼던 넘들..이젠 마땅한 처벌을 받아야지..,그래...                False   
3                    1,2화 어설펐는데 3,4화 지나서부터는 갈수록 너무 재밌던데                False   
4     1. 사람 얼굴 손톱으로 긁은것은 인격살해이고2. 동영상이 몰카냐? 메걸리안들 생각...                 True   
...                                                 ...                  ...   
7891                                      힘내세요~ 응원합니다!!                False   
7892                             힘내세요~~삼가 고인의 명복을 빕니다..                False   
7893                              힘내세용 ^^ 항상 응원합니닷 ^^ !                False   
7894  힘내소...연기로 답해요.나도 53살 인데 이런일 저런일 다 있더라구요.인격을 믿습...                False   
7895                                 힘들면 관뒀어야지 그게 현명한거다                False   

        bias  hate  
0     others  hate  
1       none  none  
2       none  hate  
3       none  none  
4     gender  hate  
...      ...   ...  
7891    none  none  
7892    none  none  
7893    none  none  
7894    none  none  
7895    none  none  

[7896 rows x 4 columns]

##### social bias : gender, others none
##### hate speech : none, offensive, hate

In [15]:
train['hate'].value_counts()

none         3486
offensive    2499
hate         1911
Name: hate, dtype: int64

In [14]:
train['bias'].value_counts()

none      5148
others    1516
gender    1232
Name: bias, dtype: int64

In [24]:
train[train['bias'].apply(lambda x: x=='gender')].contain_gender_bias.value_counts()

True    1232
Name: contain_gender_bias, dtype: int64

- gender에 대한 bias가 있는 1232명은 contain_gender_bias 값이 True로 할당되어 있음
- 즉 contain_gender_bias column은 삭제해도됨